server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeService.java

```diff
@@ -249,4 +250,12 @@ public static boolean isMigratingToRemoteStore(Metadata metadata) {
 
         return (isMixedMode && isRemoteStoreMigrationDirection);
     }
+
+    public RemoteStoreNodeStats getRemoteStoreNodeStats() {
+        long lastSuccessfulFetchOfPinnedTimestamps = 0;
+        if (RemoteStoreSettings.isPinnedTimestampsEnabled()) {
+            lastSuccessfulFetchOfPinnedTimestamps = RemoteStorePinnedTimestampService.getPinnedTimestamps().v1();
+        }
+        return new RemoteStoreNodeStats(lastSuccessfulFetchOfPinnedTimestamps);
```

Comment:
If the static method is called to initialise `RemoteStoreNodeStats`, do we really need to inject `RemoteStoreNodeService`? 

While initialising `RemoteStoreNodeStats`, we fetch the required details. When it is actually needed, we can do the wiring.